### PR TITLE
isNumber() bugfix

### PR DESCRIPTION
**Pull request by develart-projects**
Using isNumber() in current implementation is causing float to be used as array key and that's prohibited by newer PHP versions.
Here is the bugfix for the problem. Already tested localy on production servers.

### CODE REVIEW

**Comment by develart-projects** [comment]
..anyone here? :)

File: library/Zend/Locale/Format.php [hunk 1]
```diff
@@ -524,7 +528,10 @@ public static function isNumber($input, array $options = [])
 
         $regexs = Zend_Locale_Format::_getRegexForType('decimalnumber', $options);
         $regexs = array_merge($regexs, Zend_Locale_Format::_getRegexForType('scientificnumber', $options));
-        if (!empty($input) && ($input[0] == $symbols['decimal'])) {
+        
+        $firstChar = substr($input,0,1);
```
Comment (glensc) [inline library/Zend/Locale/Format.php:532]: why is this better than `$input[0]`?

**Comment by glensc** [comment]
provide simple reproducible test case. i.e php snippet. I believe this is locale aware problem, so include your locale setup details.

**Comment by develart-projects** [comment]
> provide simple reproducible test case. i.e php snippet. I believe this is locale aware problem, so include your locale setup details.

Any localization using ',' instead of '.' as decimal separator triggers the warning, when used as filter in form filed definition. (As far as I remember, because I applied this fix like year+ ago. Now just patching the upstream as well. )

**Comment by glensc** [comment]
i've found a fix for localization aware float to use sprintf %F formatter

**Comment by glensc** [comment]
you say "triggers the warning", but you have not provided what the warning is.

and still, if you want this to be merged, provide a code snippet to reproduce it. other maintainers may be more forgiving and merge without any tests, but not me.

**Comment by develart-projects** [comment]
> you say "triggers the warning", but you have not provided what the warning is.
> 
> and still, if you want this to be merged, provide a code snippet to reproduce it. other maintainers may be more forgiving and merge without any tests, but not me.

Msg was something about using float as array key is not allowed. I don't remember where exactly I found the problem, so can't really reproduce immediately. Need to run into that again.

 Trying to get UTs working right now to verify the change. Still some work to be done there, commented in another thread (UT9 migration).

**Comment by develart-projects** [comment]
I managed to find place of my code where "old" code is triggering the warning:
`Warning: Trying to access array offset on value of type float in /usr/share/NetBeansProjects/library/development/Zend/Locale/Format.php on line 527`

`/usr/share/NetBeansProjects/library/development/Zend/Locale/Format.php:527:float 400`

Now I'm trying to reproduce this problem using some real-life sample code to move this PR forward.

**Comment by develart-projects** [comment]
OK, here is how to reproduce the problem / how to trigger the warning:

`$input = 0.005; // valid numeric entry
$result = Zend_Locale_Format::isNumber($input);`

As you can see, array access to the float does not really work. My patch is fixing the problem.
Now, please re-check the PR and merge, or let me know if you like to get that fixed another way. I would like to move forward here and mainline my current modified copy.

Thanks in advance.

**Comment by develart-projects** [comment]
Anyone here or already dead?

**Pull request merged**